examples/contrib/named/tree_data.py

```diff
@@ -87,10 +91,12 @@ def main(args):
     }
 
     print('Step\tLoss')
+    loss = 0.0
     for step in range(args.num_epochs):
-        if step % 100 == 0:
-            loss = inference.step(data)
+        loss += inference.step(data)
+        if step and step % 10 == 0:
             print('{}\t{:0.5g}'.format(step, loss))
+            loss = 0.0
```

Comment:
i havent looked at this example with much detail, but why is the loss being accumulated rather than set (as it was previously)?

Reply:
I'm accumulating it until I print it, then reset the accumulator. This is to overcome noise in the loss. Previous to this PR the loss looked non-monotonic.

Reply:
(also the old logic is totally wrong: most iterations didn't even .step() 😕 )